### PR TITLE
Re-ordered plug-in result to be included _after_ chat-history

### PR DESCRIPTION
**Pull request by crickman**
### Motivation and Context

Customer reported stale plug-in results.  Re-ordering prompt to include plug-in results last may impact prioritizing this data for certain cases.  (Isolated testing indicates that the model can respond to priority hints via ordering, but other factors may influence model behavior as well...such as chat history or memory presenting stronger semantic context.)

### Description

Move planner results to after chat-history:

```
[SOURCE START]

This is the result of invoking the functions listed after "PLUGINS USED:" to retrieve additional information outside of the data you were trained on. You can use this data to help answer the user's query.

PLUGINS USED: ListPluginPlugin.GetList.

RESULT: {
"content": "eba26d18-b4bb-46d4-b6b6-cbf60d4312e8\n12ed4781-5b22-48a0-b4e9-a351e546e7ff\n90605eb2-51ce-4058-8ae0-dfbe5ed455f7\nf51d2923-f97a-41b8-97d4-991b9f46e260",
"contentType": "text/plain"
}

[SOURCE END]
```

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by alliscode — APPROVED** [review]

File: webapi/Skills/ChatSkills/ChatSkill.cs [hunk 1]
```diff
@@ -383,9 +383,10 @@ private async Task<ChatMessage> GetChatResponseAsync(string chatId, string userI
         var documentMemories = tasks[1];
 
         // Fill in the chat history if there is any token budget left
-        var chatContextComponents = new List<string>() { chatMemories, documentMemories, planResult };
+        var chatContextComponents = new List<string>() { chatMemories, documentMemories };
         var chatContextText = string.Join("\n\n", chatContextComponents.Where(c => !string.IsNullOrEmpty(c)));
-        var chatHistoryTokenLimit = remainingToken - TokenUtilities.TokenCount(chatContextText);
+        var chatHistoryTokenLimit = remainingToken - TokenUtilities.TokenCount(chatContextText) - TokenUtilities.TokenCount(planResult);
```
Comment (TaoChenOSU) [inline webapi/Skills/ChatSkills/ChatSkill.cs:388]: `ExtractChatHistoryAsync` will use all of `chatHistoryTokenLimit`, leaving no token for the plan.

Comment (crickman) [reply webapi/Skills/ChatSkills/ChatSkill.cs:388]: `TokenUtilities.TokenCount(planResult)` accounts for the remaining portion at the end.

Comment (TaoChenOSU) [reply webapi/Skills/ChatSkills/ChatSkill.cs:388]: Saw that the number of tokens the plan used was taken out. Resolving this comment.

File: webapi/Skills/ChatSkills/ChatSkill.cs [hunk 2]
```diff
@@ -395,6 +396,11 @@ private async Task<ChatMessage> GetChatResponseAsync(string chatId, string userI
             chatContextText = $"{chatContextText}\n{chatHistory}";
         }
 
+        if (!string.IsNullOrWhiteSpace(planResult))
```
Comment (TaoChenOSU) [inline webapi/Skills/ChatSkills/ChatSkill.cs:399]: nit: comment on why we are doing this

**Review by glahaye — APPROVED** [review]

**Pull request merged**